### PR TITLE
Fix logic error and make some settings optional in wg.conf template.

### PR DESCRIPTION
**Pull request by nanhantianyi**
Fix logic error and make some settings optional in wg.conf template.

### CODE REVIEW

**Comment by ngoduykhanh** [comment]
Hey @nanhantianyi thanks for the PR. Could you please make your PR description more informative to speed up my PR review process? Which was the issue and why do we need some sitting optional in the config file template?
Please also always add an empty line at the end of the file. Thank you!

**Comment by nanhantianyi** [comment]
Hi @ngoduykhanh ,there is an {{end}} tag in wrong place which make the "Endpoint" config out of the range of ".Client.Enabled".
Make “PostUp”、”PreDown“、”PostDown“、”Table“ optional to ensure there is no empty config in generated file if user don't set them.

File: templates/wg.conf [hunk 1]
```diff
@@ -7,11 +7,11 @@
...
-{{if .globalSettings.MTU}}MTU = {{ .globalSettings.MTU }}{{end}}
-PostUp = {{ .serverConfig.Interface.PostUp }}
-PreDown = {{ .serverConfig.Interface.PreDown }}
-PostDown = {{ .serverConfig.Interface.PostDown }}
-Table = {{ .globalSettings.Table }}
+{{if .globalSettings.MTU }}MTU = {{ .globalSettings.MTU }}
+{{end}}{{if .serverConfig.Interface.PostUp }}PostUp = {{ .serverConfig.Interface.PostUp }}
+{{end}}{{if .serverConfig.Interface.PreDown }}PreDown = {{ .serverConfig.Interface.PreDown }}
+{{end}}{{if .serverConfig.Interface.PostDown }}PostDown = {{ .serverConfig.Interface.PostDown }}
+{{end}}{{if .globalSettings.Table }}Table = {{ .globalSettings.Table }}{{end}}
```
Comment (ngoduykhanh) [inline templates/wg.conf:14]: If am not mistaken, wireguard allows us to have these configs set with an empty value. That's why we didn't care much about removing them completely from the config file, to make it easier to maintain the template. For that reason, I don't think we need if statement here.

File: templates/wg.conf [hunk 2]
```diff
@@ -21,7 +21,7 @@ Table = {{ .globalSettings.Table }}
 # Update at:    {{ .Client.UpdatedAt }}
 [Peer]
 PublicKey = {{ .Client.PublicKey }}
-{{if .Client.PresharedKey }}PresharedKey = {{ .Client.PresharedKey }}{{end}}
-AllowedIPs = {{$first :=true}}{{range .Client.AllocatedIPs }}{{if $first}}{{$first = false}}{{else}},{{end}}{{.}}{{end}}{{range .Client.ExtraAllowedIPs }},{{.}}{{end}}{{end}}
+{{if .Client.PresharedKey }}PresharedKey = {{ .Client.PresharedKey }}
+{{end}}AllowedIPs = {{$first :=true}}{{range .Client.AllocatedIPs }}{{if $first}}{{$first = false}}{{else}},{{end}}{{.}}{{end}}{{range .Client.ExtraAllowedIPs }},{{.}}{{end}}
```
Comment (ngoduykhanh) [inline templates/wg.conf:25]: Can you keep the original convention? I kept the `{{end}}` on the same line of each config to make it easier to read.

**Review by ngoduykhanh — CHANGES_REQUESTED** [review]
Thanks for your clarification. It is a nice catch. I left 2 comments in your PR.

File: templates/wg.conf (hunk 2 above)
Comment (nanhantianyi) [reply templates/wg.conf]: Okey，I write in this format just because there is no empty line generated if not set "PresharedKey".

**Pull request merged**